packages/@tailwindcss-upgrade/src/codemods/migrate-tailwind-directives.test.ts

```diff
@@ -0,0 +1,172 @@
+import dedent from 'dedent'
+import postcss from 'postcss'
+import { expect, it } from 'vitest'
+import { migrateTailwindDirectives } from './migrate-tailwind-directives'
+
+const css = dedent
+
+function migrate(input: string) {
+  return postcss()
+    .use(migrateTailwindDirectives())
+    .process(input, { from: expect.getState().testPath })
+    .then((result) => result.css)
+}
+
+it("should not migrate `@import 'tailwindcss'`", async () => {
+  expect(
+    await migrate(css`
+      @import 'tailwindcss';
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss';
+  `)
+})
+
+it('should migrate the default @tailwind directives to a single import', async () => {
+  expect(
+    await migrate(css`
+      @tailwind base;
+      @tailwind components;
+      @tailwind utilities;
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss';
+  `)
+})
+
+it('should migrate the default @tailwind directives as imports to a single import', async () => {
+  expect(
+    await migrate(css`
+      @import 'tailwindcss/base';
+      @import 'tailwindcss/components';
+      @import 'tailwindcss/utilities';
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss';
+  `)
+})
+
+it.each([
+  [
+    css`
+      @tailwind components;
+      @tailwind base;
+      @tailwind utilities;
+    `,
+  ],
+  [
+    css`
+      @tailwind components;
+      @tailwind utilities;
+      @tailwind base;
+    `,
+  ],
+  [
+    css`
+      @tailwind utilities;
+      @tailwind base;
+      @tailwind components;
+    `,
+  ],
+  [
+    css`
+      @tailwind utilities;
+      @tailwind components;
+      @tailwind base;
+    `,
+  ],
+])(
+  'should migrate the default directives (but in different order) to a single import, order %#',
+  async (input) => {
+    expect(await migrate(input)).toEqual(css`
+      @import 'tailwindcss';
+    `)
+  },
+)
+
+it('should migrate `@tailwind base` to theme and preflight imports', async () => {
+  expect(
+    await migrate(css`
+      @tailwind base;
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss/theme' layer(theme);
+    @import 'tailwindcss/preflight' layer(base);
+  `)
+})
+
+it('should migrate `@import "tailwindcss/base"` to theme and preflight imports', async () => {
+  expect(
+    await migrate(css`
+      @import 'tailwindcss/base';
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss/theme' layer(theme);
+    @import 'tailwindcss/preflight' layer(base);
+  `)
+})
+
+it('should migrate `@tailwind utilities` to an import', async () => {
+  expect(
+    await migrate(css`
+      @tailwind utilities;
+    `),
+  ).toEqual(css`
+    @import 'tailwindcss/utilities' layer(utilities);
+  `)
+})
+
+it('should migrate `@import "tailwindcss/utilities"` to an import', async () => {
+  expect(
+    await migrate(css`
+      @import 'tailwindcss/utilities';
```

Comment:
Did this even exist in v3? Feels weird to migrate this if it didn't.

Reply:
Yep! it's for when you are using `postcss-import` where `@import` has to be at the top, so you would replace `@tailwind utilities` with `@import "tailwindcss/utilities"`. 

See: https://tailwindcss.com/docs/functions-and-directives#config

Reply:
oh right 🤦‍♂️